src/parser/smt2/smt2.cpp

```diff
@@ -462,6 +471,101 @@ Term Smt2State::mkIndexedConstant(const std::string& name,
   return Term();
 }
 
+Term Smt2State::mkIndexedConstant(const std::string& name,
+                                  const std::vector<std::string>& symbols)
+{
+  if (d_logic.isTheoryEnabled(internal::theory::THEORY_STRINGS))
+  {
+    if (name == "char")
+    {
+      if (symbols.size() != 1)
+      {
+        parseError("Unexpected number of indices for char");
+      }
+      if (symbols[0].length() <= 2 || symbols[0].substr(0, 2) != "#x")
+      {
+        parseError(std::string("Unexpected index for char: `") + symbols[0]
+                   + "'");
+      }
+      return mkCharConstant(symbols[0].substr(2));
+    }
+  }
+  else if (d_logic.hasCardinalityConstraints())
+  {
+    if (name == "fmf.card")
+    {
+      if (symbols.size() != 2)
+      {
+        parseError("Unexpected number of indices for fmf.card");
+      }
+      Sort t = getSort(symbols[0]);
+      // convert second symbol back to a numeral
+      uint32_t ubound;
+      std::stringstream ss;
+      ss << symbols[1];
+      ss >> ubound;
```

Comment:
Don't you need to check whether the string is a valid numeral here?

Reply:
Technically, no, since parsing should guarantee this is a numeral.

I noticed that other places in the code also make this assumption, I'll update this to call the generic utility method stringToUnsigned.